packages/core/src/Cardano/types/Certificate.ts

```diff
@@ -115,6 +115,7 @@ export interface StakeAddressCertificate {
 
 export interface PoolRegistrationCertificate {
   __typename: CertificateType.PoolRegistration;
+  deposit?: Lovelace;
```

Comment:
I would prefer to have a separate type for HydratedTx, but this will do for now. Let's add a comment :)
```suggestion
  /**
   * This type is used for both `Tx` and `HydratedTx` certificates.
   * It will always be `undefined` for `Tx` when building/deserializing a transaction.
   */
  deposit?: Lovelace;
```

Reply:
Sorry @mkazlauskas , does 34347c1495 address this?

Reply:
Almost! Just need to update https://github.com/input-output-hk/cardano-js-sdk/commit/34347c1495fd032a46d76474a956939b4fcde455#diff-a4136fb38ea6bd360a39db1a56bf7429bc431c6be1c8937bcd98f58e2dc52462R89 to exclude `certificates` from base type, and use the regular `Certificate[]` instead. The inheritance is a little bit backwards there :sweat_smile:

Reply:
77461a742f should address this as well